lib/internal/test_runner/runner.js

```diff
@@ -230,12 +230,11 @@ function run(options) {
     postRun = undefined;
   }
 
-  PromisePrototypeThen(SafePromiseAllReturnVoid(testFiles, (path) => {
+  PromisePrototypeThen(SafePromiseAllSettledReturnVoid(testFiles, (path) => {
     const subtest = runTestFile(path, root, inspectPort, filesWatcher);
     runningSubtests.set(path, subtest);
     return subtest;
-  }),
-                       postRun, postRun);
+  }), postRun, postRun);
```

Comment:
nit: `%Promise.allSettled%` can never throw
```suggestion
  }), postRun);
```